frame/executive/src/lib.rs

```diff
@@ -369,28 +364,31 @@ where
 		}
 	}
 
-	fn initial_checks(block: &Block) {
-
-		sp_tracing::enter_span!(sp_tracing::Level::TRACE, "initial_checks");
+	fn ver_checks(block: &Block, public_key: Vec<u8>) {
+		// Check that `parent_hash` is correct.
+		sp_tracing::enter_span!(sp_tracing::Level::TRACE, "ver checks");
 		let header = block.header();
+		let n = header.number().clone();
+		// Check that shuffling seedght is generated properly
+		let new_seed = VRFOutput::from_bytes(&header.seed().seed.as_bytes())
+			.expect("cannot parse shuffling seed");
```

Comment:
Are these expects in wasm runtime?
Perhaps it is best to avoid panics in the runtime.
How sure are we that this will never fail?

Reply:
actually thats inteded to fail, assertions and unwraps is how frame_executive handles exceptions as its in WASM even if it panics that doesnt make host client code to panic. Similarly all the conditions are handled 
https://github.com/mangata-finance/substrate/blob/mangata-dev/frame/executive/src/lib.rs#L295